truss-build/src/truss/build/commands.py

```diff
@@ -0,0 +1,93 @@
+from abc import abstractmethod
+import shlex
+
+
+# TODO(bola): Support secrets
+class Command:
+    @abstractmethod
+    def serialize(self):
+        pass
+
+
+class FromCommand(Command):
+    def __init__(self, image, tag=None, AS=None):
+        self.image = image
+        self.tag = tag
+        self._as = AS
+
+    def serialize(self):
+        ret = f"FROM {self.image}"
+        if self.tag is not None:
+            ret += f":{self.tag}"
+        if self._as is not None:
+            ret += f" AS {self._as}"
+        return ret
+
+
+class RunCommand(Command):
+    def __init__(self, command, mounts=None):
+        self.command = command
+        self.mounts = mounts
+
+    def serialize(self):
+        cmd = f"RUN "
+        if self.mounts is not None:
+            for mount in self.mounts:
+                cmd += f"--mount={mount} "
+        cmd += self.command
```

Comment:
Nit: like previous comment, collect parts in list and join at the end. This loop has in principle quadratic runtime.